lib/scholar/linear/linear_regression.ex

```diff
@@ -0,0 +1,94 @@
+defmodule Scholar.Linear.LinearRegression do
+  @moduledoc """
+  Ordinary least squares linear regression.
+  """
+  import Nx.Defn
+
+  @derive {Nx.Container, containers: [:coefficients]}
+  defstruct [:coefficients]
+
+  opts = [
+    sample_weights: [
+      type: {:list, {:custom, Scholar.Options, :positive_number, []}}
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts)
+
+  @doc """
+  Fits a linear regression model for sample inputs `a` and
+  sample targets `b`.
+  """
+  deftransform fit(a, b, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @opts_schema)
+
+    opts =
+      [
+        sample_weights_flag: opts[:sample_weights] != nil
+      ] ++
+        opts
+
+    {sample_weights, opts} = Keyword.pop(opts, :sample_weights, 1.0)
+    sample_weights = Nx.tensor(sample_weights)
```

Comment:
`sample_weights` should be an optional argument instead.

#58 has a similar need (and solution!) for dealing with optional opts + weights

Reply:
@polvalente we agreed on making them tensors later on. So this approach is fine as is by me.

But I am also wondering if we should just allow a `Nx.Tensor` as a direct argument to `Nx.tensor`. WDYT?

Reply:
So it would be a no-op? The only problem I see is that it could be less obvious to people where they have tensors vs where they have numbers. 

Didn't we soft deprecate tensors-as-arguments to Nx.iota for a similar reason?

Reply:
We deprecated because we were passing tensors in places where shapes were allowed and in many planes shapes are options, which led to whole tensors being cached. In this case, it is tensor in, tensor out, so we should be fine.

Reply:
Alright then